src/main/java/services/gorest/stepdefinition/PostsStepsDefinition.java

```diff
@@ -0,0 +1,99 @@
+import static utils.variables.SessionVariableManager.getSessionVariable;
+import static utils.variables.SessionVariableManager.setSessionVariable;
+import static utils.variables.SessionVariables.*;
+
+public class PostsStepsDefinition {
+
+    @Steps
+    private CreatePost createPost;
+
+    @Steps
+    private GetPost getPost;
+
+    @Steps
+    private UpdatePost updatePost;
+
+    @Steps
+    private DeletePost deletePost;
+
+    @Steps
+    private ReusableMethods reusableMethods;
+
+    @Steps
+    private CreateUser createUser;
+
+    @Steps
+    private DeleteUser deleteUser;
+
+    @Before("@PostSmoke")
+    public void createPrereq() {
+        Response response = createUser.whenCreateRandomUserObject();
+        setSessionVariable(VAR_USER_ID, response.as(GetUserResponse.class).getResult().getId());
+    }
+
+    @When("^I create a new post with my user id (.*), I provide the title (.*) and add the following body:$")
+    public void whenCreatePost(String userId, String title, String body) {
+        int uId = Integer.valueOf(replaceExpectedWithVariable(userId, VAR_USER_ID));
+        Response response = createPost.whenCreateNewPost(uId, title, body);
+        GetPostResponse getPostResponse = response.as(GetPostResponse.class);
+        setSessionVariable(VAR_RESPONSE, response);
+        if (getSessionVariable(VAR_RESPONSE) != null) {
+            setSessionVariable(VAR_POST_ID, getPostResponse.getResult().getId());
+        }
+    }
+
+    @When("^I retrieve a single post based on the id: (.*)$")
+    public void whenGetPostById(String postId) {
+        replaceExpectedWithVariable(postId, VAR_POST_ID);
+        Response response = getPost.getPostById(getSessionVariable(VAR_POST_ID));
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @When("^I delete a single post based on the id: (.*)$")
+    public void whenDeletePostById(String postId) {
+        replaceExpectedWithVariable(postId, VAR_POST_ID);
+        Response response = deletePost.deletePostUsingId(getSessionVariable(VAR_POST_ID));
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+    @When("^I update a single post using my user id (.*) with the new title (.*)$")
+    public void whenUpdatePostById(String userId, String title) {
+        int uId = Integer.valueOf(replaceExpectedWithVariable(userId, VAR_USER_ID));
+        Response response = updatePost.whenUpdatePostsTitleUsingId(getSessionVariable(VAR_POST_ID), uId, title);
+        setSessionVariable(VAR_RESPONSE, response);
+    }
+
+
+    @After("@PostSmoke")
+    public void tearDownDeletePost(Scenario scenario) {
+        if (!scenario.getName().equals("Deleting post details")) {
+            Response response = getPost.getPostById(getSessionVariable(VAR_POST_ID));
+            if (response.as(GetPostResponse.class).get_meta().getCode() == 200) {
+                deletePost.deletePostUsingId(getSessionVariable(VAR_POST_ID));
```

Comment:
to be easy to use in future steps, you can create a method which checs if the user/ post/etc exists and it should return a bool